src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -307,6 +307,20 @@ private static bool IsFloatPointExpressionOfUnknownPrecision(BoundExpression rew
 
                     break;
 
+                case ConversionKind.MethodGroup:
+
+                    if (_inExpressionLambda || explicitCastInCode || (object)symbolOpt == null || !symbolOpt.IsStatic || symbolOpt.MethodKind != MethodKind.Ordinary)
+                    {
+                        break;
+                    }
+                    if (symbolOpt.IsExtensionMethod && ((BoundMethodGroup)rewrittenOperand).ReceiverOpt.Kind != BoundKind.TypeExpression)
+                    {
+                        break;
+                    }
+
+                    return RewriteMethodGroupConversion(oldNode, syntax, rewrittenOperand, symbolOpt, isExtensionMethod, rewrittenType);
```

Comment:
Figured out. `isExtensionMethod` is the value from `BoundMethodGroup.IsExtensionMethod`, while `symbolOpt.IsExtensionMethod` says whether the method is declared as extension method, `isExtensionMethod` says whether the method is used as extension method. So the checks above can be simplified. Check [here](https://github.com/diryboy/roslyn/commit/9da6164579bafa75831413da0093a3a8033255bd#diff-a5c0a608b7c2112a72aeeaa3fd45a9d2L312)